pkg/compact/compact.go

```diff
@@ -727,7 +729,7 @@ func (rs *RetentionProgressCalculator) ProgressCalculate(ctx context.Context, gr
 type Planner interface {
 	// Plan returns a list of blocks that should be compacted into single one.
 	// The blocks can be overlapping. The provided metadata has to be ordered by minTime.
-	Plan(ctx context.Context, metasByMinTime []*metadata.Meta) ([]*metadata.Meta, error)
+	Plan(ctx context.Context, metasByMinTime []*metadata.Meta) ([]compactionTask, error)
```

Comment:
I am trying to understand the interface. The return value of `compactionTasks` means those are tasks that we can run in parallel safely?

Reply:
Yes, that would be the idea. I've added this a documentation on the type.